packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -190,13 +217,13 @@ const devHtmlHook: IndexHtmlTransformHook = async (
 
   await Promise.all(
     styleUrl.map(async ({ start, end, code }, index) => {
-      const url = filename + `?html-proxy&${index}.css`
+      const url = `${proxyModulePath}?html-proxy&index=${index}.css`
 
       // ensure module in graph after successful load
       const mod = await moduleGraph.ensureEntryFromUrl(url, false)
       ensureWatchedFile(watcher, mod.file, config.root)
 
-      const result = await server!.pluginContainer.transform(code, url)
+      const result = await server!.pluginContainer.transform(code, mod.id!)
```

Comment:
The `transform` hook expects the resolved id as second parameter

Reply:
BTW, `mod.id` is `/path-to/index.html?html-proxy&index=0.css` 😀

Reply:
BTW, `mod.id` is `/path-to/index.html?html-proxy&index=0.css` 😀